core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp

```diff
@@ -198,6 +198,79 @@ int cuda_get_opt_block_size(const CudaInternal* cuda_instance,
                                 LaunchBounds{});
 }
 
+struct ExperimentalCudaLaunchParameters {
+  size_t shared_mem;
+  size_t block_size;
+};
+
+// Assuming cudaFuncSetCacheConfig(MyKernel, cudaFuncCachePreferL1)
+inline size_t get_shmem_per_sm_prefer_l1(cudaDeviceProp const& properties) {
+  int const compute_capability = properties.major * 10 + properties.minor;
+  switch (compute_capability) {
+    case 30:
+    case 32:
+    case 35: return 16;
+    case 37: return 80;
+    case 50:
+    case 53:
+    case 60:
+    case 62: return 64;
+    case 52:
+    case 61: return 96;
+    case 70:
+    case 80: return 8;
+    case 75: return 32;
+    default:
+      Kokkos::Impl::throw_runtime_exception(
+          "Unknown device in cuda block size deduction");
+  }
+  return 0;
+}
+
+// Default block size deduction
+template <class FunctorType, class Policy,
+          std::enable_if_t<!Policy::experimental_contains_desired_occupancy>* =
+              nullptr>
+inline ExperimentalCudaLaunchParameters
+experimental_cuda_get_opt_launch_configuration(
+    const CudaInternal* cuda_instance, const cudaFuncAttributes& attr,
+    const FunctorType& f, const Policy&, const size_t vector_length,
+    const size_t shmem_block, const size_t shmem_thread) {
+  using LaunchBounds = typename Policy::launch_bounds;
+  ExperimentalCudaLaunchParameters launch_params;
+  launch_params.block_size = cuda_get_opt_block_size<FunctorType, LaunchBounds>(
+      cuda_instance, attr, f, vector_length, shmem_block, shmem_thread);
+  launch_params.shared_mem = 0;
```

Comment:
why is this 0? I mean you still have shmem_block and shmem_thread right? At least we should give it a different name if its not the total shared memory ...

Reply:
This function is not used any more.  Will remove.